clients/cobol-lsp-vscode-extension/src/services/copybook/downloader/CopybookDownloader.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+import { CopybookName } from "../CopybookDownloadService";
+
+export interface CopybookDownloader {
+  /**
+   * Downloads a file based on the passed parameters
+   *
+   * @param copybookName Copybook to be downloaded.
+   * @param documentUri cobol programs which needs copybook
+   * @param addressLocator locator of the remote copybook.
+   */
+  downloadCopybook(
+    copybookName: CopybookName,
+    documentUri: string,
+    addressLocator: unknown,
```

Comment:
doesn't the `unknown` defeat the purpose of having a common interface?

Reply:
Initial thought was that its really unknown, A FTPDownloader might need a {hostname, port} , A Endeavour Downloader might need some endeavour  related settings etc.

Reply:
I thought as much... however that requires the caller to know what is being called, which kind of defeats the purpose of hiding the implementation behind an interface.

Reply:
updated